### PR TITLE
ci: consolidate geoviewer job, include epic_ in detector_config json

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
We create a list of detector configs in an early step and use it later on. This expands that functionality to create both a csv and json list. Both have now `epic_<configuration>` as format, consistent with other uses of `DETECTOR_CONFIG`.

This PR also avoids a large number of CI jobs on eicweb to trigger many geoviewers: each CI job was building the entire geometry again. We can avoid this, and just create the geoviewer artifacts in one go. We pass the csv list for that reason.

We also turn the matrix jobs for fast operations back into sequential jobs, avoiding the repeated overhead of starting a container environment.

TODO:
- [x] merge https://eicweb.phy.anl.gov/EIC/benchmarks/geoviewer/-/merge_requests/2 if success
- [x] modify back to using `main` instead of `detector-config-csv-list`

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

### CODE REVIEW

File: .github/workflows/linux-eic-shell.yml [hunk 1]
```diff
@@ -182,7 +171,11 @@ jobs:
         network_types: "none"
         setup: install/setup.sh
         run: |
-          checkGeometry -c ${DETECTOR_PATH}/${DETECTOR}_${{ matrix.detector_config }}.xml
+          configs=${{ needs.list-detector-configs.outputs.configs_csv }}
+
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:175]:
```suggestion
```

File: .github/workflows/linux-eic-shell.yml [hunk 2]
```diff
@@ -132,43 +137,27 @@ jobs:
...
-      matrix: ${{ fromJson(needs.list-detector-configs.outputs.configs) }}
     steps:
     - uses: eic/trigger-gitlab-ci@v2
       id: trigger
       with:
         url: https://eicweb.phy.anl.gov
         project_id: 539
         token: ${{ secrets.EICWEB_GEOVIEWER_TRIGGER }}
-        ref_name: main
+        ref_name: detector-config-csv-list
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:147]:
```suggestion
        ref_name: main
```

File: .github/workflows/linux-eic-shell.yml [hunk 3]
```diff
@@ -223,20 +213,21 @@ jobs:
...
-          geoConverter -compact2gdml -input ${DETECTOR_PATH}/${DETECTOR}_${{ matrix.detector_config }}.xml -output ${{ matrix.detector_config }}.gdml
+          configs=${{ needs.list-detector-configs.outputs.configs_csv }}
+          for config in ${configs//,/ } ; do
+            geoConverter -compact2gdml -input ${DETECTOR_PATH}/${config}.xml -output ${config}.gdml
+          done
     - uses: actions/upload-artifact@v3
       with:
         name: gdml
-        path: ${{ matrix.detector_config }}.gdml
+        path: *.gdml
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:223]:
```suggestion
        path: ${{ github.workspace }}/*.gdml
```

File: .github/workflows/linux-eic-shell.yml [hunk 4]
```diff
@@ -250,11 +241,14 @@ jobs:
...
-          geoConverter -compact2tgeo -input ${DETECTOR_PATH}/${DETECTOR}_${{ matrix.detector_config }}.xml -output ${{ matrix.detector_config }}.root
+          configs=${{ needs.list-detector-configs.outputs.configs_csv }}
+          for config in ${configs//,/ } ; do
+            geoConverter -compact2tgeo -input ${DETECTOR_PATH}/${config}.xml -output ${config}.root
+          done
     - uses: actions/upload-artifact@v3
       with:
         name: tgeo
-        path: ${{ matrix.detector_config }}.root
+        path: *.root
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:251]:
```suggestion
        path: ${{ github.workspace }}/*.root
```

File: .github/workflows/linux-eic-shell.yml [hunk 5]
```diff
@@ -182,7 +171,10 @@ jobs:
         network_types: "none"
         setup: install/setup.sh
         run: |
-          checkGeometry -c ${DETECTOR_PATH}/${DETECTOR}_${{ matrix.detector_config }}.xml
+          configs=${{ needs.list-detector-configs.outputs.configs_csv }}
+          for config in ${configs//,/ } ; do
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:175]:
```suggestion
          for config in $(echo ${configs//,/ }) ; do
```

File: .github/workflows/linux-eic-shell.yml [hunk 6]
```diff
@@ -223,20 +213,21 @@ jobs:
         network_types: "none"
         setup: install/setup.sh
         run: |
-          geoConverter -compact2gdml -input ${DETECTOR_PATH}/${DETECTOR}_${{ matrix.detector_config }}.xml -output ${{ matrix.detector_config }}.gdml
+          configs=${{ needs.list-detector-configs.outputs.configs_csv }}
+          for config in ${configs//,/ } ; do
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:217]:
```suggestion
          for config in $(echo ${configs//,/ }) ; do
```

File: .github/workflows/linux-eic-shell.yml [hunk 7]
```diff
@@ -250,11 +241,14 @@ jobs:
         network_types: "none"
         setup: install/setup.sh
         run: |
-          geoConverter -compact2tgeo -input ${DETECTOR_PATH}/${DETECTOR}_${{ matrix.detector_config }}.xml -output ${{ matrix.detector_config }}.root
+          configs=${{ needs.list-detector-configs.outputs.configs_csv }}
+          for config in ${configs//,/ } ; do
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:245]:
```suggestion
          for config in $(echo ${configs//,/ }) ; do
```

File: .github/workflows/linux-eic-shell.yml [hunk 8]
```diff
@@ -182,7 +171,10 @@ jobs:
         network_types: "none"
         setup: install/setup.sh
         run: |
-          checkGeometry -c ${DETECTOR_PATH}/${DETECTOR}_${{ matrix.detector_config }}.xml
+          configs=${{ needs.list-detector-configs.outputs.configs_csv }}
+          for config in $(echo ${configs//,/ }) ; do
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:175]: Ok, `IFS` then.
```suggestion
          IFS=, for config in ${configs//,/ } ; do
```

**Comment by wdconinc** [comment]
Cleans up nicely.

Screenshot_20230708-115442

File: .github/workflows/linux-eic-shell.yml [hunk 9]
```diff
@@ -132,43 +137,28 @@ jobs:
...
-      matrix: ${{ fromJson(needs.list-detector-configs.outputs.configs) }}
     steps:
     - uses: eic/trigger-gitlab-ci@v2
       id: trigger
       with:
         url: https://eicweb.phy.anl.gov
         project_id: 539
         token: ${{ secrets.EICWEB_GEOVIEWER_TRIGGER }}
-        ref_name: main
+        ref_name: post-pr-comment
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml:147]:
```suggestion
        ref_name: main
```

File: .github/workflows/linux-eic-shell.yml [hunk 10]
```diff
@@ -223,20 +216,23 @@ jobs:
...
+          for config in ${configs[@]} ; do
+            echo "::group::${config}" ;
+            geoConverter -compact2gdml -input ${DETECTOR_PATH}/${config}.xml -output ${config}.gdml ;
+            echo "::endgroup::" ;
+          done
     - uses: actions/upload-artifact@v3
       with:
         name: gdml
-        path: ${{ matrix.detector_config }}.gdml
+        path: ${{ github.workspace }}/*.gdml
```
Comment (veprbl) [inline .github/workflows/linux-eic-shell.yml:228]: Why not just?
```suggestion
        path: *.gdml
```

**Review by veprbl — DISMISSED** [review]

File: .github/workflows/linux-eic-shell.yml (hunk 10 above)
Comment (wdconinc) [reply .github/workflows/linux-eic-shell.yml:228]: Not accepted in https://github.com/eic/epic/actions/runs/5494997781 with annotation in https://github.com/eic/epic/actions/runs/5494997781/workflow

Comment (wdconinc) [reply .github/workflows/linux-eic-shell.yml:228]: The problem isn't that it doesn't parse the `*` without quotes; it does when there is an explicit path prefix. Maybe with `./*.gdml`, but is that so much better?

Comment (veprbl) [reply .github/workflows/linux-eic-shell.yml:228]: It's better in the sense that it doesn't refer to `github.workspace` consistently.

Comment (wdconinc) [reply .github/workflows/linux-eic-shell.yml:228]:
```suggestion
        path: "*.gdml"
```

**Review by veprbl — APPROVED** [review]

File: .github/workflows/linux-eic-shell.yml [hunk 11]
```diff
@@ -250,11 +246,16 @@ jobs:
...
+          for config in ${configs[@]} ; do
+            echo "::group::${config}" ;
+            geoConverter -compact2tgeo -input ${DETECTOR_PATH}/${config}.xml -output ${config}.root
+            echo "::endgroup::" ;
+          done
     - uses: actions/upload-artifact@v3
       with:
         name: tgeo
-        path: ${{ matrix.detector_config }}.root
+        path: ${{ github.workspace }}/*.root
```
Comment (wdconinc) [inline .github/workflows/linux-eic-shell.yml]:
```suggestion
        path: "*.root"
```

**Pull request merged**